pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -0,0 +1,769 @@
+        self._http_proxy_revealer = self.builder.get_object("http_proxy_revealer")
+        self._http_proxy_location_entry = self.builder.get_object("http_proxy_location_entry")
+        self._http_proxy_username_entry = self.builder.get_object("http_proxy_username_entry")
+        self._http_proxy_password_entry = self.builder.get_object("http_proxy_password_entry")
+
+        # RHSM baseurl
+        self._custom_rhsm_baseurl_checkbox = self.builder.get_object(
+            "custom_rhsm_baseurl_checkbox"
+        )
+        self._custom_rhsm_baseurl_revealer = self.builder.get_object(
+            "custom_rhsm_baseurl_revealer"
+        )
+        self._custom_rhsm_baseurl_entry = self.builder.get_object(
+            "custom_rhsm_baseurl_entry"
+        )
+
+        # server hostname
+        self._custom_server_hostname_checkbox = self.builder.get_object(
+            "custom_server_hostname_checkbox"
+        )
+        self._custom_server_hostname_revealer = self.builder.get_object(
+            "custom_server_hostname_revealer"
+        )
+        self._custom_server_hostname_entry = self.builder.get_object(
+            "custom_server_hostname_entry"
+        )
+
+        # status label
+        self._registration_status_label = self.builder.get_object("registration_status_label")
+
+        # register button
+        self._register_button = self.builder.get_object("register_button")
+
+        # * the subscription status tab * #
+
+        # general status
+        self._method_status_label = self.builder.get_object("method_status_label")
+        self._role_status_label = self.builder.get_object("role_status_label")
+        self._sla_status_label = self.builder.get_object("sla_status_label")
+        self._usage_status_label = self.builder.get_object("usage_status_label")
+        self._insights_status_label = self.builder.get_object("insights_status_label")
+
+        # attached subscriptions
+        self._attached_subscriptions_label = self.builder.get_object(
+            "attached_subscriptions_label"
+        )
+        self._subscriptions_listbox = self.builder.get_object("subscriptions_listbox")
+
+        # unregister button
+        self._unregister_revealer = self.builder.get_object("unregister_revealer")
+        self._unregister_button = self.builder.get_object("unregister_button")
+
+        # setup spoke state based on data from the Subscription DBus module
+        self._update_spoke_state()
+
+        # Fixup Insights defaults
+        #
+        # FIXME: could be done better in module if module can see if kickstart was used
+        #        for the given installation and set InsightsEnabled to True
+        #
+        # If we have a kickstart, use the value from the module, otherwise enable Insights
+        # as that's the default for interactive installations.
+        if flags.automatedInstall:
+            self._insights_checkbox.set_active(self._subscription_module.InsightsEnabled)
```

Comment:
This is already set up in `_update_spoke_state`. I think you should set the DBus property `InsightsEnabled` to `True` for interactive installations before you call `_update_spoke_state` (ideally at the beginning of the `initialize` or `__init__` method).

Reply:
Nice solution, let's do that! :)